Controller/PageAdminController.php

```diff
@@ -117,12 +117,14 @@ public function createAction(Request $request = null)
             $sites = $this->get('sonata.page.manager.site')->findBy(array());
 
             if (count($sites) == 1) {
-                return $this->redirect($this->admin->generateUrl('create', array(
+                $options = array_merge(array(
                     'siteId' => $sites[0]->getId(),
                     'uniqid' => $this->admin->getUniqid(),
-                )));
-            }
+                ), $request->query->all());
 
+                return $this->redirect($this->admin->generateUrl('create', $options));
```

Comment:
If you do choose to use the `+` operator, maybe revert back to the version without the `$options` one-time variable?